espflash/src/cli/mod.rs

```diff
@@ -47,6 +47,9 @@ pub struct ConnectArgs {
     /// Serial port connected to target device
     #[arg(short = 'p', long, env = "ESPFLASH_PORT")]
     pub port: Option<String>,
+    /// Require confirmation before auto-connecting to a recognized device.
+    #[arg(short = 'i', long)]
+    pub interactive: bool,
```

Comment:
Do you mind moving these arguments to espflash/src/bin/espflash.rs and cargo-espflash/main.rs? It would require some duplication but, changing the CLI mod would result in a breaking change requiring us to bump the version to 3.0

Reply:
Ah, good point about the compatibility issue. Unfortunately, I don't see how to get around it. I can add an `interactive` flag in various places, but eventually I need to pass the `interactive` flag to the (public) `cli::connect` function. I can't add it to the `ConnectArgs` or `Config` struct because they're both public, and adding it as an extra parameter would also break semver :disappointed: 

I guess I could deprecate `cli::connect` and add a new version that takes an `interactive` argument... but that starts to feel like overkill. I think the better solution would be to add some `#[non_exhaustive]` annotations so that arguments are easier to add in the future, but of course adding those annotations will itself be a breaking change.

Reply:
Yes, as you mentioned, there are plans for using `#[non_exhaustive]` (https://github.com/esp-rs/espflash/issues/473), not sure if we can work around this, or we just need to wait to #473 to be resolved and add this to `v3`

Reply:
Did a bit more of investigation and couldn't find a workaround for this, I'll add the `v3` milestone, and we can merge it once we start working towards `v3`.